modules/polyfills/src/fetch-node/read-file.node.js

```diff
@@ -30,9 +30,15 @@ export async function readFile(url, options = {}) {
 
   if (isRequestURL(url)) {
     return new Promise((resolve, reject) => {
-      options = {...new URL(url), ...options};
+      const _url = new URL(url);
+      const _options = {
+        hostname: _url.hostname,
+        path: _url.pathname,
+        method: 'GET',
+        headers: options.headers || {}
```

Comment:
Since our request implementation now always accept the content encodings, we could just inject those three encoding headers into the headers list here inside the request call. I am pretty sure that happens in the fetch case in the browser. The app just requests a file and does not care if it was compressed.

Reply:
I'll see if this could be made this way. Making it transparent is a good idea. Will check few other comments and see how it works with huge files (if it's chunked).